lib/archethic/replication.ex

```diff
@@ -108,17 +106,14 @@ defmodule Archethic.Replication do
             %{role: :chain}
           )
 
-          if previous_tx do
-            # Load the rest of the chain asynchronously
-            Task.start(fn ->
-              # Stream the insertion of the chain
-              previous_tx
-              |> stream_previous_chain()
-              |> Stream.reject(&Enum.empty?/1)
-              |> Stream.each(&TransactionChain.write/1)
-              |> Stream.run()
-            end)
-          end
+          # Stream the insertion of the chain
+          tx
+          |> stream_previous_chain()
+          |> Stream.reject(&Enum.empty?/1)
+          |> Stream.each(&TransactionChain.write/1)
+          |> Stream.run()
+
+          TransactionChain.write_transaction(tx)
```

Comment:
telemetry should be calculated after DB writing no ?

Reply:
in fact this telemetry it's about the replication validation.
But I can add one for the full write